db/sequences.c

```diff
@@ -0,0 +1,192 @@
+/*
+   Copyright 2017 Bloomberg Finance L.P.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+#include "sequences.h"
+#include <int_overflow.h>
+
+extern struct dbenv *thedb;
+
+int gbl_sequence_replicant_distribution = 0;
+
+/**
+ *  Returns the next value for a specified sequence object. If the sequence name
+ *  cannot be found, -1 is returned. If the next value is greater than the max
+ *  or less than min value, for ascending or descending sequences respectively,
+ *  and cycle is not enabled, <error>.
+ *
+ *  @param name char * Name of the sequence
+ *  @param val long long * Reference to output location
+ */
+int seq_next_val(tran_type *tran, char *name, long long *val)
+{
+    sequence_t *seq = getsequencebyname(name);

```

Comment:
Grab berkley level readlock on 'name' before this call.  This might be a slightly larger project than I was thinking as we'll have to come up with a locking convention for sequences and add this type to our print-lock utilities.